runtimes/dotnet-6.0/src/Program.cs

```diff
@@ -3,69 +3,163 @@
 using System.IO;
 using System.Text;
 using System.Text.Json;
+using Newtonsoft.Json;
+using System.Web;
 
 var app = WebApplication.Create(args);
 app.Urls.Add("http://0.0.0.0:3000");
-app.MapPost("/", Execute);
+app.MapMethods("/{*path}", new[] { "GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS", "HEAD", "TRACE" }, Execute);
 app.Run();
 
-static async Task<IResult> Execute(
-    [FromHeader(Name = "x-internal-challenge")] string? challengeId,
-    [FromBody] RuntimeRequest? request)
+static async Task<IResult> Execute(HttpRequest Request)
 {
-    if (challengeId is not null 
-        && challengeId != Environment.GetEnvironmentVariable("INTERNAL_RUNTIME_KEY"))
+    string Secret = Request.Headers.TryGetValue("x-open-runtimes-secret", out var SecretValue) ? (string) SecretValue : "";
+    if(Secret == "" || Secret != (Environment.GetEnvironmentVariable("OPEN_RUNTIMES_SECRET") ?? ""))
     {
-        return Results.Problem(
-            detail: "Unauthorized",
-            statusCode: 500);
+        return new CustomResponse("Unauthorized. Provide correct \"x-open-runtimes-secret\" header.", 500);
```

Comment:
Can we use the built-in `Results` instead of a `CustomResponse` class?

Reply:
I wasnt able to find a way to use Results with custom body, status code and headers. That's why Implemented this one.